libflux/go/libflux/analyze_test.go

```diff
@@ -56,3 +58,170 @@ func TestAnalyze(t *testing.T) {
 		})
 	}
 }
+
+func TestMergePackages(t *testing.T) {
+	pkg1 := libflux.Parse(`
+package foo
+
+a = 1`)
+	pkg2 := libflux.Parse(`
+package foo
+
+b = 1`)
+	err := libflux.MergePackages(pkg1, pkg2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gotFB, err := pkg2.MarshalFB()
+	got := ast.DeserializeFromFlatBuffer(gotFB)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &ast.Package{
+		Package: "foo",
+		Files: []*ast.File{
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "b"},
+						Init: &ast.IntegerLiteral{Value: 1},
+					},
+				},
+			},
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "a"},
+						Init: &ast.IntegerLiteral{Value: 1},
+					},
+				},
+			},
+		},
+	}
+
+	if !cmp.Equal(got, want, asttest.IgnoreBaseNodeOptions...) {
+		t.Errorf(
+			"MergePackages unexpected packages -want/+got:\n %v",
+			cmp.Diff(want, got, asttest.IgnoreBaseNodeOptions...),
+		)
+	}
+}
+
+func TestNoInFiles(t *testing.T) {
+	pkg1 := libflux.Parse(`package foo`)
+	pkg2 := libflux.Parse(`
+package foo
+
+b = 1`)
+
+	err := libflux.MergePackages(pkg1, pkg2)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	gotFB, err := pkg2.MarshalFB()
+	got := ast.DeserializeFromFlatBuffer(gotFB)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	want := &ast.Package{
+		Package: "foo",
+		Files: []*ast.File{
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+				Body: []ast.Statement{
+					&ast.VariableAssignment{
+						ID:   &ast.Identifier{Name: "b"},
+						Init: &ast.IntegerLiteral{Value: 1},
+					},
+				},
+			},
+			{
+				Name:     "",
+				Metadata: "parser-type=rust",
+				Package: &ast.PackageClause{
+					Name: &ast.Identifier{Name: "foo"},
+				},
+			},
+		},
+	}
+
+	if !cmp.Equal(got, want, asttest.IgnoreBaseNodeOptions...) {
+		t.Errorf(
+			"MergePackages unexpected packages -want/+got:\n %v",
+			cmp.Diff(want, got, asttest.IgnoreBaseNodeOptions...),
+		)
+	}
+}
+
+func TestNoOutFiles(t *testing.T) {
+	pkg1 := libflux.Parse(`
```

Comment:
There's some boilerplate here that's repeated for each test method, you could use an array of test cases and loop over them to be more concise.

Reply:
Yes,  I did try that but it was causing an issue in the test information. I wasn't sure if conciseness was worth spending time on another bug.

Reply:
Nevermind. Fixed this.